File: backend/src/main/scala/microservices/operations/hotel/api/RegisterHotelManagerPlanner.scala
// RegisterHotelManagerPlanner 负责operations相关实现。

package com.typesafe.travel.operations.domain

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.auth.domain.hashPasswordForLoginEmail
import com.typesafe.travel.persistence.operations.HotelManagerPlainSql
import com.typesafe.travel.shared.kernel.EmailAddress

import java.sql.Connection
import java.time.Instant
import java.util.UUID

object RegisterHotelManagerPlanner extends ConnectionApiPlan[RegisterHotelManagerPlannerRequest, ManagerSessionPlannerResponse]:
  override val name: String = "RegisterHotelManagerPlanner"
  override def plan(input: RegisterHotelManagerPlannerRequest, connection: Connection): IO[ManagerSessionPlannerResponse] =
    val now = Instant.now()
    val hotelId = s"hotel-${UUID.randomUUID().toString.take(12)}"
    val managerId = s"manager-${UUID.randomUUID().toString.take(12)}"
    for
      email <- IO.fromEither(EmailAddress.create(input.email))
      _ <- validateRegisterHotel(input)
      passwordHash <- hashPasswordForLoginEmail(input.password, email)
      _ <- HotelManagerPlainSql.insertHotel(connection, hotelId, input.hotelName, input.location, now)
      _ <- HotelManagerPlainSql.insertHotelManager(connection, managerId, hotelId, input.email, input.displayName, now)
      _ <- HotelManagerPlainSql.insertHotelManagerCredential(connection, managerId, input.email, passwordHash, now)
    yield ManagerSessionPlannerResponse(managerId, "Hotel", input.email, input.displayName, "Active", hotelId, None, now.toString)

File: frontend/src/microservices/operations/hotel/api/RegisterHotelManagerPlanner.ts
// 本文件定义 RegisterHotelManagerPlanner，负责 operations 模块的注册编排和接口入口。

import { executeJsonApiRequest } from '@/shared-kernel/api/ApiTransport'
import type { RegisterHotelManagerPlannerRequest } from '@/microservices/operations/hotel/objects/RegisterHotelManagerPlannerRequest'
import type { HotelManagerSessionPlannerResponse } from '@/microservices/operations/hotel/objects/HotelManagerSessionPlannerResponse'

export const registerHotelManager = (payload: RegisterHotelManagerPlannerRequest): Promise<HotelManagerSessionPlannerResponse> =>
  executeJsonApiRequest<HotelManagerSessionPlannerResponse>('/RegisterHotelManagerPlanner', 'POST', payload)
